lib/backends/sdc/utils.js

```diff
@@ -5,32 +5,71 @@
  */
 
 /*
- * Copyright (c) 2015, Joyent, Inc.
+ * Copyright 2016, Joyent, Inc.
  */
 
 var assert = require('assert-plus');
+var format = require('util').format;
+var mod_fwrule = require('fwrule');
 var net = require('net');
 var restify = require('restify');
 
 var common = require('../../common');
 var constants = require('../../constants');
 var Link = require('../../models/link');
+var triton_tags = require('../../triton-tags');
 
-var mod_fwrule = require('fwrule');
 
 
 function vmUuidToShortDockerId(uuid) {
     return uuid.replace(/-/g, '').slice(0, 12);
 }
 
+/**
+ * Docker containers are stored on the VM "tags" field
+ * (https://github.com/joyent/sdc-vmapi/blob/master/docs/index.md#vm-metadata).
+ * General docker labels are prefixed with 'docker:label:' (AFAIK the only
+ * need for that separation is the use of the 'sdc_docker' tag for fwrules).
+ *
+ * In addition, the special structured 'triton.'-prefixed tags are exposed
+ * as Docker labels (see DOCKER-728).
+ *
+ * Note: Docker label values must be strings otherwise the docker 1.10 client
+ * will blow up. See https://github.com/docker/docker/issues/20881 and
+ * DOCKER-721. Currently VM.tags are limited to string, boolean or number,
+ * so we can simply `.toString()`.
+ */
 function dockerLabelsFromVmTags(tags) {
     var labels = {};
-    var labelPrefix = common.LABELTAG_PREFIX;
+    const LABELTAG_PREFIX = common.LABELTAG_PREFIX;
+    const expectedTypes = {string: true, boolean: true, number: true};
```

Comment:
easier lookup that arr.indexOf, basically. I don't have a string pref.
If JS had Python's `in` operator, I'd use an array for this kind of thing much more frequently.